tests/php/Dev/Tasks/LegacyThumbnailMigrationHelperTest.php

```diff
@@ -68,21 +67,48 @@ public function tearDown()
         parent::tearDown();
     }
 
+    /**
+     * Tests cases where files have already been migrated to 4.3 or older,
+     * and some images have since been unpublished or otherwise marked protected.
+     * The previously orphaned thumbnails wouldn't have been moved to the protected store in this case.
+     * The migration task should find those orphans, and reunite them with the original image.
+     * And they lived happily ever after.
+     */
     public function testMigratesThumbnailsForProtectedFiles()
     {
         /** @var TestAssetStore $store */
         $store = singleton(AssetStore::class); // will use TestAssetStore
 
+        // Remove the legacy helper, otherwise it'll move the _resampled files as well,
+        // which "pollutes" our test case.
+        /** @var FileIDHelperResolutionStrategy $strategy */
+        $publicStrategy = $store->getPublicResolutionStrategy();
```

Comment:
Hm, I wanted to specifically test with the defaults, *apart* from the `LegacyFileIDHelper`. So if those defaults ever change, that change should be tested through this unit test. So I think my way is more accurate :)

Reply:
There's a separate angle to this discussion: How much of what we call "unit tests" (ha ha ha) should rely on global configuration which can change on a project-by-project basis. That's currently enshrined through `$backend->setPublicResolutionStrategy(Injector::inst()->get(FileResolutionStrategy::class . '.public'));` in `TestAssetStore` though, so I'm not keen to change it at this late stage since it'll affect all kinds of tests.